webextension/js/confirm-page.js

```diff
@@ -1,31 +1,85 @@
-const redirectUrl = new URL(window.location).searchParams.get("url");
-document.getElementById("redirect-url").textContent = redirectUrl;
-const redirectSite = new URL(redirectUrl).hostname;
-document.getElementById("redirect-site").textContent = redirectSite;
+async function load() {
+  const searchParams = new URL(window.location).searchParams;
+  const redirectUrl = decodeURIComponent(searchParams.get("url"));
+  const cookieStoreId = searchParams.get("cookieStoreId");
+  const currentCookieStoreId = searchParams.get("currentCookieStoreId");
+  const redirectUrlElement = document.getElementById("redirect-url");
+  redirectUrlElement.textContent = redirectUrl;
+  createFavicon(redirectUrl, redirectUrlElement);
 
-document.getElementById("redirect-form").addEventListener("submit", (e) => {
-  e.preventDefault();
+  const container = await browser.contextualIdentities.get(cookieStoreId);
+  [...document.querySelectorAll(".container-name")].forEach((containerNameElement) => {
+    containerNameElement.textContent = container.name;
+  });
+
+  // If default container, button will default to normal HTML content
+  if (currentCookieStoreId) {
+    const currentContainer = await browser.contextualIdentities.get(currentCookieStoreId);
+    document.getElementById("current-container-name").textContent = currentContainer.name;
+  }
+
+  document.getElementById("redirect-form").addEventListener("submit", (e) => {
+    e.preventDefault();
+    const buttonTarget = e.explicitOriginalTarget;
+    switch (buttonTarget.id) {
+    case "confirm":
+      confirmSubmit(redirectUrl, cookieStoreId);
+      break;
+    case "deny":
+      denySubmit(redirectUrl);
+      break;
+    }
+  });
+}
+
+function createFavicon(pageUrl, redirectUrlElement) {
+  const origin = new URL(pageUrl).origin;
+  const imageElement = document.createElement("img");
+  imageElement.src = `${origin}/favicon.ico`;
+
+  redirectUrlElement.prepend(imageElement);
```

Comment:
Not really... however we could just say "Test pilot is an experiment... yada" which we have for other issues like this and fix when stable. I already got involved in discussion about getting a way to query the favicon store in extensions instead.
We could load it in a throw away container that is never visible etc but I'm not sure it's actually solving the privacy issue. (I think I mentioned it in the issue were @ChenMorpheus wanted the title too)